### ISSUE TITLE
Git submodule command failed on raspberry-pi

### ISSUE THREAD

**Issue by schneefisch** [andig/homebridge-fritz#50]
Hello,

I am trying to install homebridge and homebridge-fritz on a raspberry-pi 3
I did try with different os versions (ubuntu mate, raspbian, raspbian-stretch) and I always get following error: 

```
sudo npm install -g --unsafe-perm homebridge-fritz

npm ERR! code 128
npm ERR! Command failed: /usr/bin/git submodule update -q --init --recursive
npm ERR! fatal: Could not change back to '/root/.npm/_cacache/tmp/git-clone-412c949d': Keine Berechtigung
npm ERR!
```

That might be due to the fact, that the most current git-version on debian is 2.11.0:

```
$ git --version
git version 2.11.0
```

Anyway, do you know that problem? Is there a solution? 

The detailed logs are:

```
25 http fetch GET 304 https://registry.npmjs.org/fritzapi 1295ms (from cache)
26 silly pacote range manifest for fritzapi@^0.10.4 fetched in 1302ms
27 silly resolveWithNewModule fritzapi@0.10.4 checking installable status
28 silly fetchPackageMetaData error for fritzbox@git+https://github.com/mapero/Fritzbox.git Command failed: /usr/bin/git submodule update -q --init --recursive
28 silly fetchPackageMetaData fatal: Could not change back to '/root/.npm/_cacache/tmp/git-clone-412c949d': Keine Berechtigung
29 timing stage:rollbackFailedOptional Completed in 9ms
30 timing stage:runTopLevelLifecycles Completed in 5686ms
31 verbose stack Error: Command failed: /usr/bin/git submodule update -q --init --recursive
31 verbose stack fatal: Could not change back to '/root/.npm/_cacache/tmp/git-clone-412c949d': Keine Berechtigung
31 verbose stack
31 verbose stack     at ChildProcess.exithandler (child_process.js:276:12)
31 verbose stack     at emitTwo (events.js:126:13)
31 verbose stack     at ChildProcess.emit (events.js:214:7)
31 verbose stack     at maybeClose (internal/child_process.js:915:16)
31 verbose stack     at Socket.stream.socket.on (internal/child_process.js:336:11)
31 verbose stack     at emitOne (events.js:116:13)
31 verbose stack     at Socket.emit (events.js:211:7)
31 verbose stack     at Pipe._handle.close [as _onclose] (net.js:561:12)
32 verbose cwd /home/pi/git
33 verbose Linux 4.14.50-v7+
34 verbose argv "/usr/bin/node" "/usr/bin/npm" "install" "-g" "--unsafe-perm" "homebridge-fritz"
35 verbose node v8.12.0
36 verbose npm  v6.4.1
37 error code 128
38 error Command failed: /usr/bin/git submodule update -q --init --recursive
38 error fatal: Could not change back to '/root/.npm/_cacache/tmp/git-clone-412c949d': Keine Berechtigung
39 verbose exit [ 1, true ]
```

**Comment by andig** [comment]
Interesting. Looks like a problem with https://github.com/mapero/Fritzbox.git which is not published on npm. Unfortunately I really don't know what could be the problem here :(

**Comment by schneefisch** [comment]
hmm, thanks.

**Comment by LoPablo** [comment]
Maybe this helps. Had the same error and this worked for me:

I cloned this repo into /usr/lib/node_modules with

```
cd /usr/lib/node_modules
git clone https://github.com/andig/homebridge-fritz.git
```

Afterwards I ran
```
npm install -g homebridge-fritz
```
again and this time it refused to install because the homebridge-fritz folder in /usr/lib/node_modules was a git repo.
```
pi@pihome:/usr/lib/node_modules/homebridge-fritz# npm install -g homebridge-fritz
npm ERR! path /usr/lib/node_modules/homebridge-fritz
npm ERR! code EISGIT
npm ERR! git /usr/lib/node_modules/homebridge-fritz: Appears to be a git repo or submodule.
npm ERR! git     /usr/lib/node_modules/homebridge-fritz
npm ERR! git Refusing to remove it. Update manually,
npm ERR! git or move it out of the way first.
```

Removed the cloned folder via 
```
rm -rf homebridge-fritz
```
and ran npm again.
Everything went well and it installed.

**Comment by schneefisch** [comment]
Thank you,

I got it now running with Ubuntu server 16.04
It's now working although very often has timeouts on the iPad... 

But it's working for a start ;)

**Comment by TheNavigator86** [comment]
I got the same error using a very similar setup (raspberry-pi 2, raspbian).
There is an issue attached to npm that helped me find a way to install homebridge-fritz on raspbian: https://github.com/npm/npm/issues/16898#issuecomment-336644312
Although it is only a workaround and no "clean" solution, it is possible to perform the installation by switching to root and performing `npm install -g --unsafe-perm homebridge-fritz` as the root user.

**Comment by m-rossi** [comment]
None of the hints above worked for me and my Raspberry. I got it working by using `sudo su -` before upgrading/installing the package. Source: https://github.com/npm/npm/issues/18038#issuecomment-319654959

**Comment by andig** [comment]
This should be fixed by 5c8a5d96f1e38858fa65fad098f1dc5e03305db1

**Comment by andig** [comment]
Fixed by #62
